src/bookmarks/background/addition.js

```diff
@@ -50,9 +51,20 @@ export async function createNewPageForBookmark(id, bookmarkInfo) {
     }
 }
 
-export async function createBookmarkByExtension(url) {
+export async function createBookmarkByExtension(url, tabId) {
+    console.log(url, tabId)
     const pageId = generatePageDocId({ url })
-    const pageDoc = await db.get(pageId)
+    let pageDoc
+    try {
+        pageDoc = await db.get(pageId)
+    } catch (err) {
+        try {
+            await logPageVisit({ tabId, url })
```

Comment:
Maybe better to make `logPageVisit` either return the page doc it creates or pass in a flag to tell it to make a bookmark doc as well? It will make a visit doc by default, but in the case of a quick bookmark then close tab, I don't think that should be made. 

Maybe a separate interface function on the same level as `logPageVisit` which shares the page creation logic may be better